packages/realm/src/Object.ts

```diff
@@ -58,14 +61,22 @@ const DEFAULT_PROPERTY_DESCRIPTOR: PropertyDescriptor = { configurable: true, en
 /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
 const PROXY_HANDLER: ProxyHandler<RealmObject<any>> = {
   ownKeys(target) {
-    return Reflect.ownKeys(target).concat([...target.keys()].map(String));
```

Comment:
Just curious what `Reflect.ownKeys(target)` would return that `target[KEY_ARRAY]` does not?

Reply:
Anything that is directly set on the target. This includes our internal symbols, and anything else that the user added (hopefully only symbols, but they can also do `this.not_a_realm_field = 1;`

Reply:
Also, I think that unless the user does an `Object.defineProperty()` to forcibly replace a realm field (or accidentally ends up using class fields which do this implicitly...), `Reflect.ownKeys(target)` and `target[KEY_ARRAY]` should be completely disjoint, since we set the realm fields as properties on the prototype chain rather than on the object itself so they don't count as "own keys".

Reply:
Ah ok...so `target[KEY_ARRAY]` would be the defined properties and `Reflect.ownKeys(target)` returns user added fields.  Thanks!